sklearn/utils/tests/test_stats.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+
+from sklearn.utils.stats import _weighted_percentile
+
+
+def test_weighted_percentile():
+    y = np.empty(102, dtype=np.float64)
+    y[:50] = 0
+    y[-51:] = 2
+    y[-1] = 100000
+    y[50] = 1
+    sw = np.ones(102, dtype=np.float64)
+    sw[-1] = 0.0
+    score = _weighted_percentile(y, sw, 50)
+    assert score == 1
```

Comment:
if we have a floating point number, just use `pytest.approx(...)` here and after.